lib/pbench/server/api/__init__.py

```diff
@@ -100,6 +100,7 @@ def register_endpoints(api, app, config):
     )
     api.add_resource(
         DatasetsInventory,
+        f"{base_uri}/datasets/inventory/<string:dataset>/",
         f"{base_uri}/datasets/inventory/<string:dataset>/<path:path>",
         endpoint="datasets_inventory",
         resource_class_args=(config, logger),
```

Comment:
@dbutenhof, I've lost track:  is the trailing slash supposed to be there or are we supposed to omit it (i.e., at line 103)?

Reply:
In this case, we could probably go with the three rules again and expose the tarball with or without the trailing slash. Either should result in `path == None`, which is what's important here.

In `/datasets/contents/<dataset>/` the trailing `/` is used to imply we're looking at the root directory. That's not a reasonable target for `/datasets/inventory` so there's no ambiguity... but maybe it'd make sense to allow `/datasets/inventory/<dataset>` here to make that more visually clear. (And maybe even drop the trailing slash form?)

Reply:
😆 So, @dbutenhof, what should I be asking @siddardh-ra to change, here?  (You said "could", "maybe it'd make sense to", and "maybe even drop"...so...what should we do??  🙃)

Reply:
I would list this as an option; I'm OK with adding the slash version if you like that idea, but I don't see any need to hold the PR on that unless it's important to you. (We can always add it later.)

Reply:
Well, given that this PR is awaiting the changes that you've requested and that Peter has seconded, I would be happy to see this addressed in that work.

So, @siddardh-ra, I believe the directive is, add a third rule:
```
    api.add_resource(
        DatasetsInventory,
        f"{base_uri}/datasets/inventory/<string:dataset>",
        f"{base_uri}/datasets/inventory/<string:dataset>/",
        f"{base_uri}/datasets/inventory/<string:dataset>/<path:path>",
        endpoint="datasets_inventory",
        resource_class_args=(config, logger),
```

Reply:
That's fine. If you add the extra URI rule, Siddardh, you should add a test case to explicitly verify that `/datasets/inventory/<dataset>` and `/datasets/inventory/<dataset>/` behave the same, perhaps somewhat like the `test_get_all` case I put in `test_server_configuration.py` with `@pytest.mark.parametrize("key", (None, ""))`